runtime/ISO_Fortran_binding.cc

```diff
@@ -193,20 +199,150 @@ int CFI_is_contiguous(const CFI_cdesc_t *descriptor) {
   return 1;
 }
 
+static inline bool IsAssumedSize(const CFI_cdesc_t *dv) {
+  return (dv->rank > 0 && dv->dim[dv->rank - 1].extent == -1);
+}
+
 int CFI_section(CFI_cdesc_t *result, const CFI_cdesc_t *source,
     const CFI_index_t lower_bounds[], const CFI_index_t upper_bounds[],
     const CFI_index_t strides[]) {
-  return CFI_INVALID_DESCRIPTOR;  // TODO
+  CFI_index_t extent[CFI_MAX_RANK];
+  CFI_index_t actualStride[CFI_MAX_RANK];
+  CFI_rank_t resRank{0};
+  char *shiftedBaseAddr{reinterpret_cast<char *>(source->base_addr)};
```

Comment:
I think reinterpret_cast should only be used when absolutely necessary. In this case static_cast is fine.

Reply:
yes indeed, thanks.